### PR TITLE
Releases/0.6.0 beta

### PR DESCRIPTION
**Pull request by FrazerBayley**
### Fixed 
- Participant's now initialize with the client instead of getting reused like before.
- Stack traces are now logged upon exceptions.
### Added
- Added audience filters, allowing for filtering of the participant through user attributes.
- AllocationStore now requires a unique user id to be given with the allocation to be stored.

### CODE REVIEW

File: ascend-sdk/src/main/java/ai/evolv/Audience.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        Map<String, Function> operatorsMap = new HashMap<>();
+        operatorsMap.put("exists", (object, a) -> ((Map<String, String>) object)
+                .containsKey(((JsonElement) a).getAsString()));
+        operatorsMap.put("kv_contains", (object, params) -> {
+            String storedValue = ((Map<String, String>) object).get(
+                    ((JsonArray) params).get(0).getAsString());
+            if (storedValue == null) {
+                return false;
+            }
+            return storedValue.contains(((JsonArray) params).get(1).getAsString());
```
Comment (teolisitza) [inline ascend-sdk/src/main/java/ai/evolv/Audience.java:30]: Can the .get(1) call ever return null?

File: ascend-sdk/src/main/java/ai/evolv/ExecutionQueue.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 class ExecutionQueue {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ExecutionQueue.class);
+    private static Logger LOGGER = LoggerFactory.getLogger(ExecutionQueue.class);
```
Comment (teolisitza) [inline ascend-sdk/src/main/java/ai/evolv/ExecutionQueue.java]: Just curious, why did you have to remove `final`?

**Review by teolisitza — APPROVED** [review]
LGTM!

**Review by teolisitza — APPROVED** [review]

File: ascend-sdk/src/main/java/ai/evolv/Audience.java (hunk 1 above)
Comment (FrazerBayley) [reply ascend-sdk/src/main/java/ai/evolv/Audience.java:30]: If it did there would be something wrong with the audience query payload. kv_contains always has an array that describes its values

File: ascend-sdk/src/main/java/ai/evolv/ExecutionQueue.java (hunk 2 above)
Comment (FrazerBayley) [reply ascend-sdk/src/main/java/ai/evolv/ExecutionQueue.java]: No reason, it was accidentally copied over. Ill revert it.

**Pull request merged**